src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -0,0 +1,1113 @@
+            //         private required int PrivateProperty { get; set; }
+            Diagnostic(ErrorCode.ERR_RequiredMembersCannotBeLessVisibleThanContainingType, "PrivateProperty").WithArguments("Outer.ProtectedClass.PrivateProperty", "Outer.ProtectedClass").WithLocation(58, 30),
+            // (63,40): error CS9503: Required member 'Outer.ProtectedClass.PrivateProtectedField' cannot be less visible than the containing type 'Outer.ProtectedClass'.
+            //         private protected required int PrivateProtectedField;
+            Diagnostic(ErrorCode.ERR_RequiredMembersCannotBeLessVisibleThanContainingType, "PrivateProtectedField").WithArguments("Outer.ProtectedClass.PrivateProtectedField", "Outer.ProtectedClass").WithLocation(63, 40),
+            // (64,30): error CS9503: Required member 'Outer.ProtectedClass.PrivateField' cannot be less visible than the containing type 'Outer.ProtectedClass'.
+            //         private required int PrivateField;
+            Diagnostic(ErrorCode.ERR_RequiredMembersCannotBeLessVisibleThanContainingType, "PrivateField").WithArguments("Outer.ProtectedClass.PrivateField", "Outer.ProtectedClass").WithLocation(64, 30),
+            // (73,30): error CS9503: Required member 'Outer.PrivateProtectedClass.PrivateProperty' cannot be less visible than the containing type 'Outer.PrivateProtectedClass'.
+            //         private required int PrivateProperty { get; set; }
+            Diagnostic(ErrorCode.ERR_RequiredMembersCannotBeLessVisibleThanContainingType, "PrivateProperty").WithArguments("Outer.PrivateProtectedClass.PrivateProperty", "Outer.PrivateProtectedClass").WithLocation(73, 30),
+            // (79,30): error CS9503: Required member 'Outer.PrivateProtectedClass.PrivateField' cannot be less visible than the containing type 'Outer.PrivateProtectedClass'.
+            //         private required int PrivateField;
+            Diagnostic(ErrorCode.ERR_RequiredMembersCannotBeLessVisibleThanContainingType, "PrivateField").WithArguments("Outer.PrivateProtectedClass.PrivateField", "Outer.PrivateProtectedClass").WithLocation(79, 30)
+        );
+    }
+
+    [Fact]
+    public void RequiredMembersCannotBeExplicitInterfaceImplementations()
+    {
+        var comp = CreateCompilationWithRequiredMembers(@"
+interface I
+{
+    int Prop { get; set; }
+}
+class C : I
+{
+    required int I.Prop { get; set; }
+}
+");
+
+        comp.VerifyDiagnostics(
+            // (8,20): error CS0106: The modifier 'required' is not valid for this item
+            //     required int I.Prop { get; set; }
+            Diagnostic(ErrorCode.ERR_BadMemberFlag, "Prop").WithArguments("required").WithLocation(8, 20)
+        );
+    }
+
+    [Fact]
+    public void UsingRequiredMemberAttributeExplicitly()
+    {
+        var comp = CreateCompilationWithRequiredMembers(@"
+using System.Runtime.CompilerServices;
+[RequiredMember]
```

Comment:
Should we demonstrate the behavior when 'RequiredMemberAttribute' is explicitly declared, allows all targets, and is placed on an unexpected symbol such as a method?

I don't think it's important to have diagnostics in that scenario, but it might be worth recording the behavior.